torch/onnx/_internal/exporter/_fx_passes.py

```diff
@@ -45,3 +45,49 @@ def remove_assertion_nodes(graph_module: torch.fx.GraphModule) -> torch.fx.Graph
             graph_module.graph.erase_node(node)
     graph_module.recompile()
     return graph_module
+
+
+def insert_flatten_between_transpose_and_view(
+        exported_program: torch.export.ExportedProgram,
+) -> torch.export.ExportedProgram:
+    """Modifies the module inplace to insert a node 'flatten' between a node 'transpose' followed by a node 'view'.
+
+    The modification takes place inplace.
+
+    Remove after issue https://github.com/pytorch/pytorch/issues/136543 is fixed.
+    """
+    modified = False
+    graph = exported_program.graph_module.graph
+    for node in graph.nodes:
+        if (node.op != "call_method" or node.target != "transpose") and (
+            node.op != "call_function"
+            or not hasattr(node.target, "name")
+            or node.target.name() != "aten::transpose.int"
+        ):
+            continue
+        insert = False
+        for user in node.users:
+            if (user.op == "call_method" and user.target == "view") or (
```

Comment:
exported program graphs will not have call_method nodes

```suggestion
            if (
```

Reply:
I applied ruff format <filename>. I did not choose this. The linter does not seem to complain.

Reply:
I meant remove the call method case. Only call function is meaningful

Reply:
Are you sure? When are they removed? If run_decompositions is removing them, then we need to keep that check.

Reply:
Yes: this is specified: https://pytorch.org/docs/stable/export.ir_spec.html#graph and https://pytorch.org/docs/stable/export.ir_spec.html#node

- 0 or more nodes of op type placeholder
- 0 or more nodes of op type call_function
- exactly 1 node of op type output
- (and get_attr)

Since the graph we get is from an ExportedProgram, we do not expect a call_methods node. It just so happens that the way run_decompositions work can handle call_methods as well.